### PR TITLE
[cli] add command `uptime` to cli

### PR DESCRIPTION
**Pull request by zhanglongxia**
This PR adds a command `uptime` to print how long the device has been running.

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -3767,6 +3769,34 @@ otError Interpreter::ProcessTxPower(uint8_t aArgsLength, char *aArgs[])
     return error;
 }
 
+otError Interpreter::ProcessUptime(uint8_t aArgsLength, char *aArgs[])
+{
+    OT_UNUSED_VARIABLE(aArgs);
+
+    otError  error = OT_ERROR_NONE;
+    uint32_t now;
+    uint32_t seconds;
```
Comment (simonlingoogle) [inline src/cli/cli.cpp:3778]:
```suggestion
    uint8_t seconds;
```

`uint8_t` is enough for 0 ~ 59 ?

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -3767,6 +3769,34 @@ otError Interpreter::ProcessTxPower(uint8_t aArgsLength, char *aArgs[])
...
+    uint32_t minutes;
```
Comment (simonlingoogle) [inline src/cli/cli.cpp:3779]:
```suggestion
    uint8_t minutes;
```
ditto

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -3767,6 +3769,34 @@ otError Interpreter::ProcessTxPower(uint8_t aArgsLength, char *aArgs[])
...
+    uint32_t hours;
```
Comment (simonlingoogle) [inline src/cli/cli.cpp:3780]:
```suggestion
    uint8_t hours;
```
ditto

File: src/cli/cli.cpp [hunk 4]
```diff
@@ -3767,6 +3769,34 @@ otError Interpreter::ProcessTxPower(uint8_t aArgsLength, char *aArgs[])
...
+    now = static_cast<uint32_t>((otPlatTimeGet() - mStartTimeUs) / 1000000);
+
+    seconds = now % 60;
+    now     = now / 60;
+    minutes = now % 60;
+    now     = now / 60;
+    hours   = now % 24;
+    days    = now / 24;
+
+    OutputLine("%02d:%02d:%02d up %d days", hours, minutes, seconds, days);
```
Comment (simonlingoogle) [inline src/cli/cli.cpp:3794]:
```suggestion
    OutputLine("%dd%dh%dm%ds", days, hours, minutes, seconds);
```
nit: formats like `3d20h30m59s` is easier to parse?

File: src/cli/README.md [hunk 5]
```diff
@@ -1938,6 +1939,16 @@ Print all ports from the allowed unsecured port list.
 Done
 ```
 
+### uptime
+
+Print how long the device has been running.
```
Comment (simonlingoogle) [inline src/cli/README.md:1944]: nit: 
```suggestion
Print how long the device has been running since started or reset.
```

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Review by bukepo — COMMENTED** [review]
If this is for debugging, should we provide a option to condition out this feature?

**Comment by simonlingoogle** [comment]
> If this is for debugging, should we provide a option to condition out this feature?

We can use `OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` if this is for debugging only. 
But personally I think this is a good command to have for all CLI devices.

**Comment by zhanglongxia** [comment]
@bukepo @simonlingoogle The command "uptime" is used to check how long the device has started or reset. Sometimes, the device is accidentally reset. We can use this command to check this case. All platforms are able to support it. I thought it is a common command,  it may be a good choice to enable it on all platforms. Comments?

**Pull request closed without merging**